File: backend/app/Models/Topic.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasOne;
use Illuminate\Database\Eloquent\SoftDeletes;

class Topic extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'user_id',
        'subject_id',
        'name',
        'study_time_minutes',
        'track_questions',
    ];

    protected function casts(): array
    {
        return [
            'track_questions' => 'boolean',
            'study_time_minutes' => 'integer',
        ];
    }

    public function subject(): BelongsTo
    {
        return $this->belongsTo(Subject::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function questionStat(): HasOne
    {
        return $this->hasOne(QuestionStat::class);
    }
}

File: mobile/native-ios/EduMindAI/Models/Topic.swift
import Foundation

struct Topic: Decodable, Identifiable {
    let id: Int
    let userId: Int?
    let subjectId: Int?
    let name: String
    let studyTimeMinutes: Int?
    let trackQuestions: Bool?
    let createdAt: String?
    let updatedAt: String?
    let questionStat: QuestionStat?
}

struct TopicFormPayload: Encodable {
    let name: String
    let trackQuestions: Bool
}

struct TopicUpdatePayload: Encodable {
    let trackQuestions: Bool?
    let name: String?
}

struct AddTopicTimePayload: Encodable {
    let minutes: Int
}
